### PR TITLE
fix bfv hps

### LINKED ISSUE #280: Multiplicative depth limitation for HPS, HPSPOVERQ and HPSOVERQLEVELED modes of BFV
If I change the multiplicative technique from `HPS` to `HPSPOVER` or `HPSPOVERQLEVELED`, the example at https://github.com/openfheorg/openfhe-integer-examples/blob/master/src/strsearch_enc_1.cpp stops working correctly.  The `BEHZ` mode works fine.

### PR DESCRIPTION
**Pull request by kimandrik**
fixes #280
fix bug in ScaleAndRound in dcrtpoly:
- Previously double was converted to integer, but if double > 2^63 there was an overflow.

### CODE REVIEW

**Review by yspolyakov — APPROVED** [review]

**Comment by kimandrik** [comment]
@dsuponitskiy-duality can you please wait with this issue, I need to check again the logic. the code does not work every 10th time for level = 67 and every 2 time for level = 66

**Comment by yspolyakov** [comment]
@kimandrik @dsuponitskiy-duality I just tried the unit tests in the `dev` branch (without the fix), and the unit tests work there, too. So the unit tests do not capture the bug.

**Comment by kimandrik** [comment]
Fixed another bug: in ScaleAndRound method, when alpha is calculated, it can be bigger than modulus, but previously it was added using ModAddFast method, instead of ModAdd method.

**Review by yspolyakov — APPROVED** [review]

File: src/core/include/lattice/hal/default/dcrtpoly-impl.h [hunk 1]
```diff
@@ -1934,38 +1941,63 @@ DCRTPolyImpl<VecType> DCRTPolyImpl<VecType>::ScaleAndRound(
             nu += tOSHatInvModsDivsFrac[i] * xi.ConvertToInt();
         }
 
-        NativeInteger alpha = static_cast<uint64_t>(nu);
+        if (nu < maxDouble) {
```
[the remark this answers is not in the document]
Comment (yspolyakov) [reply src/core/include/lattice/hal/default/dcrtpoly-impl.h]: Changed to the standard logic for handling 64-bit overflows

**Review by dsuponitskiy — APPROVED** [review]
Looks good now

**Pull request merged**